Docs/StyleGuide/Style-Guide.md

```diff
@@ -0,0 +1,287 @@
+with a blank line in-between.
+
+```dafny
+import opened StandardLibrary
+import opened Native
+
+import opened Donut
+import Coffee
+```
+
+Although not required, it's recommended to keep the order of `import`s and `include`s alphabetically, except when it
+makes more sense to group them logically.
+
+## Indentation and Line Breaks
+
+### Tabs or Spaces?
+Spaces are preferred over tabs. Tabs should only be used to remain consistent with the existing code with tabs.
+
+Use 4 spaces for each indentation.
+
+### Maximum Character Limit
+Although there is no strict requirement, it is generally recommended to have a maximum of 120 characters per line.
+
+### Functions, Methods, Predicates, and Lemmas
+Every Dafny method has the following signature.
+```dafny
+method {:<attributes>} MethodName(param1: Type, param2: Type) returns (ret: Type)
+    decreases param1
+    requires P()
+    modifies param2
+    ensures Q()
+```
+
+When possible, put MethodName and the `returns` statement on the same line, as the keyword `returns` is distinct from
+other method specification clauses, such as `decreases`, `requires`, `modifies`, and `ensures`, which should appear in
+this order. Each method specification clause should be on a separate line.
+
+In case the Method signature is too long, we can break it down.
+```dafny
+method {:<attributes>} MethodName(param1: Type, param2: Type,
+        param3: Type, param4: Type, param5: Type)
+    returns (ret1: Type, ret2: Type, ret3: Type, ret4: Type,
+        ret5: Type)
+    decreases param1
+    requires P1()
+    requires P2()
+    requires P3()
+    modifies param2
+    modifies param3
+    ensures Q1()
+    ensures Q2()
+```
+
+Multiple `requires` or `ensures` can be combined into one:
+```dafny
+requires
+    && P1()
+    && P2()
+    && P3()
+```
+The same rules apply to `function`, `predicate`, and `lemma` definitions.
+
+## Things to Avoid
+
+### Parentheses
+
+In a lot of cases, Dafny does not require parentheses around expressions. Here are some examples.
+
+* If-Else-While Statements
+```dafny
+// YES
+var i := 1;
+while i < 10 {
+    ...
+    if 1 < i {
+        ...
+    }
+    ...
+}
+
+// NO
+var i := 1;
+while (i < 10) {
+    ...
+    if (1 < i) {
+        ...
+    }
+    ...
+}
+```
+
+* Simple Boolean/Arithmetic Expressions
+```dafny
+// YES
+method Collatz(num: nat)
+    decreases *
+{
+    var n := num;
+    while 1 < n
+        decreases *
+    { 
+        n := if n % 2 == 0 then n / 2 else n * 3 + 1;
+    }
+}
+
+// NO
+method Collatz(num: nat)
+    decreases *
+{
+    var n := num;
+    while (1 < n) // unnecessary parentheses
+        decreases *
+    { 
+        n := if ((n % 2) == 0) then (n / 2) else ((n * 3) + 1); // unnecessary parentheses
+    }
+}
+```
+### Whitespace
+
+Avoid unnecessary whitespace inside expressions.
+
+#### Type Declaration
+A type declaration should have a form of `variableName: variableType`.
```

Comment:
Yes, if the type is included at all. It's also okay to leave out the colon and the type, provided Dafny can infer the type.